vdom/src/com/vdom/core/CardImplIntrigue.java

```diff
@@ -121,6 +143,42 @@ private void copperSmith(MoveContext context) {
         context.coppersmithsPlayed++;
     }
     
+    private void courtier(Game game, MoveContext context, Player currentPlayer) {
+        if (currentPlayer.getHand().size() == 0) return;
+
+        Card card = currentPlayer.controlPlayer.courtier_cardToReveal(context);
+        if (card == null) {
+            Util.playerError(currentPlayer, "Courtier errror, reveal random card.");
+            card = Util.randomCard(currentPlayer.hand);
+        }
+
+        int numTypes = card.getNumberOfTypes(currentPlayer);
+        if (numTypes > 0) {
```

Comment:
If the number of types is >= 4 (e.g. Dame Josephine) you should set the options Array to all 4 options without asking the player.